blockchain/blockchain.go

```diff
@@ -1359,7 +1363,10 @@ func (bc *BlockChain) writeStateTrie(block *types.Block, state *state.StateDB) e
 				}
 			}
 			bc.lastCommittedBlock = block.NumberU64()
-			bc.chPrune <- block.NumberU64()
+
+			if bc.IsLivePruningRequired() {
+				bc.chPrune <- block.NumberU64()
+			}
```

Comment:
If pruning is available on archive mode as well, this condition appears duplicated and can be consolidated within the if {} else {} block below.

Reply:
Yes, they could be consolidated, but having the two also looks pretty because both are below the "bc.lastCommittedBlock = block.NumberU64()" line.

Reply:
Ah, Keep it this. In the `else` branch, there's another guard `if isCommitTrieRequired(bc, block.NumberU64()) {`, which I overlooked it.